ballerina/channel_csv_io.bal

```diff
@@ -52,43 +52,67 @@ isolated function channelWriteCsv(string path, FileWriteOption option, string[][
         if option == APPEND {
             string[] headersFromCSV = check readHeadersFromCsvFile(path);
             headersFromCSV = check validateCsvHeaders(headersFromCSV, headers);
-            return headersFromCSV.length() > 0 ? writeRecordsToCsvFile(path, contentToWrite, headersFromCSV, option, true) 
-            : writeRecordsToCsvFile(path, contentToWrite, headers, option, false);
+            return headersFromCSV.length() > 0 ? writeRecordsToCsvFile(path, contentToWrite, headersFromCSV, option, true)
+                : writeRecordsToCsvFile(path, contentToWrite, headers, option, false);
         }
         check writeRecordsToCsvFile(path, contentToWrite, headers, option, false);
     }
     return;
 }
 
-isolated function channelWriteCsvFromStream(WritableChannel writableChannel, stream<string[]|map<anydata>, Error?> csvStream) returns
+isolated function channelWriteCsvFromStream(string path, FileWriteOption option, stream<string[]|map<anydata>, Error?> csvStreamToWrite) returns
 Error? {
-    WritableCSVChannel csvChannel = check getWritableCSVChannel(writableChannel);
-    do {
-        if csvStream is stream<string[], Error?> {
-            record {|string[] value;|}|Error? csvRecordString = csvStream.next();
+    if csvStreamToWrite is stream<string[], Error?> {
+        WritableCSVChannel csvChannel = check getWritableCSVChannel(check openWritableCsvFile(path, option = option));
+        do {
+            record {|string[] value;|}|Error? csvRecordString = csvStreamToWrite.next();
             while csvRecordString is record {|string[] value;|} {
                 check csvChannel.write(csvRecordString.value);
-                csvRecordString = csvStream.next();
+                csvRecordString = csvStreamToWrite.next();
+            }
+        } on fail Error err {
+            check csvStreamToWrite.close();
+            check csvChannel.close();
+            return err;
+        }
+        check csvChannel.close();
+    } else if csvStreamToWrite is stream<map<anydata>, Error?> {
+        WritableCSVChannel csvChannel = check getWritableCSVChannel(check openWritableCsvFile(path, option = option));
```

Comment:
Is it correct to create a channel here? what if there is nothing to write in the stream?